# 💉​💊​ Prédiction du Risque de Diabète par Analyse Clinique et Modèles d'Apprentissage Automatique


## Les critères cliniques tels que : 
- glycémie (Glucose)
- la pression artérielle (Blood Pressure)
- l’épaisseur du pli cutané (Skin Thickness)
- l’insuline (Insulin)
- l’Indice de Masse Corporelle (BMI)
- la fonction de prédisposition génétique au diabète (Diabetes Pedigree Function) 
- l’âge (Age).

## ​✅​ Objective
* Entraîner et déployer un modèle d’intelligence artificielle capable de classer les patients selon leur risque de diabète 
* Regrouper (clustering) les données brutes afin d’identifier des profils ou comportements similaires au sein de la population étudiée.



# 📚​ Bibliothèques

In [1]:
import pandas as pd
import numpy as np


#  1️⃣ Etape 1: Chargement des données

In [9]:
# Importer les données à l’aide de pandas. Vérifier le type et la structure des colonnes (df.info() et df.head()).
df = pd.read_csv('dataset.csv')
#Affichage de 5 premier ligne
df.head()


Unnamed: 0  Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin  \
0           0            6      148             72             35        0   
1           1            1       85             66             29        0   
2           2            8      183             64              0        0   
3           3            1       89             66             23       94   
4           4            0      137             40             35      168   

    BMI  DiabetesPedigreeFunction  Age  
0  33.6                     0.627   50  
1  26.6                     0.351   31  
2  23.3                     0.672   32  
3  28.1                     0.167   21  
4  43.1                     2.288   33

In [20]:
#Analyser les types de colonnes et dimensions 
print(df.info())


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 768 entries, 0 to 767
Data columns (total 9 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Unnamed: 0                768 non-null    int64  
 1   Pregnancies               768 non-null    int64  
 2   Glucose                   768 non-null    int64  
 3   BloodPressure             768 non-null    int64  
 4   SkinThickness             768 non-null    int64  
 5   Insulin                   768 non-null    int64  
 6   BMI                       768 non-null    float64
 7   DiabetesPedigreeFunction  768 non-null    float64
 8   Age                       768 non-null    int64  
dtypes: float64(2), int64(7)
memory usage: 54.1 KB
None


# ⭐​ Conclusion :

Ce dataset contient 768 entrées complètes (sans valeurs manquantes) avec des types numériques (int et float), idéales pour l'analyse et la modélisation.

In [22]:
# le nombre de collonne et des ligne 
print(f'la data contiene {df.shape[1]} Collone et {df.shape[0]} ligne')

la data contiene 9 Collone et 768 ligne
